mk/genlink-config.mk

```diff
@@ -18,17 +18,61 @@
 ##
 
 ifeq ($(DEVICE),)
-$(error no DEVICE specified for linker script generator)
+$(warning no DEVICE specified for linker script generator)
 endif
 
 LDSCRIPT	= $(DEVICE).ld
 
-GENLINK_DEFS	:=$(shell awk -v PAT="$(DEVICE)" -v MODE="DEFS" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
-GENLINK_ARCH	:=$(shell awk -v PAT="$(DEVICE)" -v MODE="ARCH" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
-GENLINK_LIB	:=$(shell awk -v PAT="$(DEVICE)" -v MODE="LIB" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
+genlink_family		:=$(shell awk -v PAT="$(DEVICE)" -v MODE="FAMILY" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
+genlink_subfamily	:=$(shell awk -v PAT="$(DEVICE)" -v MODE="SUBFAMILY" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
+genlink_cpu		:=$(shell awk -v PAT="$(DEVICE)" -v MODE="CPU" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
+genlink_fpu		:=$(shell awk -v PAT="$(DEVICE)" -v MODE="FPU" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
+genlink_cppflags	:=$(shell awk -v PAT="$(DEVICE)" -v MODE="CPPFLAGS" -f $(OPENCM3_DIR)/scripts/genlink.awk $(OPENCM3_DIR)/ld/devices.data 2>/dev/null)
 
-DEFS		+= $(GENLINK_DEFS)
-ARCH_FLAGS	:= $(GENLINK_ARCH)
-OPENCM3_LIBNAME	:= $(strip $(subst -l,,$(GENLINK_LIB)))
+CPPFLAGS	+= $(genlink_cppflags)
 
-GENFILES	+= $(LDSCRIPT)
+ARCH_FLAGS	:=-mcpu=$(genlink_cpu)
+ifeq ($(genlink_cpu),$(filter $(genlink_cpu),cortex-m0 cortex-m0plus cortex-m3 cortex-m4 cortex-m7))
+ARCH_FLAGS    +=-mthumb
+endif
+
+ifeq ($(genlink_fpu),soft)
+ARCH_FLAGS	+= -msoft-float
+else ifeq ($(genlink_fpu),hard-fpv4-sp-d16)
+ARCH_FLAGS	+= -mfloat-abi=hard -mfpu=fpv4-sp-d16
+else ifeq ($(genlink_fpu),hard-fpv5-sp-d16)
+ARCH_FLAGS      += -mfloat-abi=hard -mfpu=fpv5-sp-d16
+else
+$(warning No match for the FPU flags)
```

Comment:
Looking at it again I recall being very cautious about breaking builds that would otherwise work. There are more places where warnings could as well be interpreted as errors in the devices.data file. It's also imaginable that the user actually intended to mix and match things. Therefore I opted for outputting warnings in all cases where I could remotely imagine the configuration to be working for now.

I'd like to revisit those warnings once it's clear if and how the linkerscript generation and target configuration by devices.data is actually used.